### PR TITLE
Documentation in the Suggestions Database

### LINKED ISSUE #1735: Documentation in the Suggestions Database
### Summary
A placeholder issue for the results of #1698.

### Value

TBC

### Specification

TBC - Discuss with @iamrecursion @4e6 

### Acceptance Criteria & Test Cases

TBC

### PR DESCRIPTION
**Pull request by BinarySoftware**
### Pull Request Description
Closes #1735.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/main/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/main/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guides.
- [x] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/main/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/main/docs/style-guide/yaml.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

**Comment by 4e6** [comment]
PR only implements the storing of HTML documentation in the suggestion database. 

In general, `documentation_html` filed in the suggestions database is not enough for IDE to show the docs. IDE needs to show documentation chunks for the module, preserving the order in which functions are defined in the source file. Entries of the suggestions database are not ordered. One way to fix that is to save the order in the `suggestions_order` table (#1698) and send it to IDE. Another way would be implementing a request that would return documentation chunks of the module in the correct order. It depends on what are IDE requirements for this task.

To proceed, someone from IDE should specify this (#1735) issue.

**Comment by iamrecursion** [comment]
Please label your PR @BinarySoftware.

**Comment by 4e6** [comment]
When updating the SQL schema, don't forget to bump the current version.
https://github.com/enso-org/enso/blob/0b363e3e85215aa0468f7ce8c17882f60f9284d9/lib/scala/searcher/src/main/scala/org/enso/searcher/sql/Tables.scala#L256-L257

File: engine/runtime/src/test/scala/org/enso/compiler/test/context/SuggestionBuilderTest.scala [hunk 1]
```diff
@@ -433,7 +450,10 @@ class SuggestionBuilderTest extends CompilerTest {
               ),
               selfType      = "Unnamed.Test.MyAtom",
               returnType    = "Number",
-              documentation = Some(" My bar")
+              documentation = Some(" My bar"),
+              documentationHtml = Some(
+                "<html><body><div class=\"doc\" style=\"font-size: 13px;\"><div><div class=\"\"><p>My bar</p></div></div></div></body></html>"
```
Comment (4e6) [inline engine/runtime/src/test/scala/org/enso/compiler/test/context/SuggestionBuilderTest.scala]: Please make a helper function that hides `<html><body>...` template. Should look like 
```
htmlDoc("<p>My bar</p>")
```

File: engine/runtime/src/main/scala/org/enso/compiler/context/SuggestionBuilder.scala [hunk 2]
```diff
@@ -147,13 +148,14 @@ final class SuggestionBuilder[A: IndexedSource](val source: A) {
...
-      returnType    = buildReturnType(returnTypeDef),
-      documentation = doc
+      externalId        = externalId,
+      module            = module.toString,
+      name              = name.name,
+      arguments         = methodArgs,
+      selfType          = selfType.toString,
+      returnType        = buildReturnType(returnTypeDef),
+      documentation     = doc,
+      documentationHtml = doc.map(DocsGenerator.runOnPureDoc)
```
Comment (4e6) [inline engine/runtime/src/main/scala/org/enso/compiler/context/SuggestionBuilder.scala:158]: If the generator function can potentially throw exceptions, they should be handled here.

Comment (BinarySoftware) [reply engine/runtime/src/main/scala/org/enso/compiler/context/SuggestionBuilder.scala:158]: It doesn't really throw much

Comment (4e6) [reply engine/runtime/src/main/scala/org/enso/compiler/context/SuggestionBuilder.scala:158]: Ok, I take it as "it doesn't throw"

**Review by 4e6 — APPROVED** [review]

**Review by kustosz — APPROVED** [review]

**Pull request merged**